src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -849,17 +850,17 @@ public function addUserProviderFactory(UserProviderFactoryInterface $factory)
     /**
      * {@inheritdoc}
      */
-    public function getXsdValidationBasePath()
+    public function getXsdValidationBasePath(): string|false
```

Comment:
We don't need to be certain of that: the caller needs to handle all possible return types anyway, that's the contracts.
We could restrict, but the only thing this would provide is to forbid child classes from returning false.
This reasoning applies to all similar situations.

Reply:
In that case the `UsernamePasswordTokenInterface` return type for the `FormLoginAuthenticator` below must be reverted :)

Reply:
It depends: do we want child classes of `FormLoginAuthenticator` to be forced to return a `UsernamePasswordToken`? The current `@return` annotation means yes we do :)

Reply:
See #42648